inc/class-core-sitemaps-provider.php

```diff
@@ -283,7 +283,7 @@ public function get_sitemap_lastmod( $name, $page ) {
 
 			// Don't schedule a duplicate job.
 			if ( ! wp_next_scheduled( 'core_sitemaps_calculate_lastmod', $event_args ) ) {
-				wp_schedule_single_event( time() + 500, 'core_sitemaps_calculate_lastmod', $event_args );
+				wp_schedule_single_event( time(), 'core_sitemaps_calculate_lastmod', $event_args );
```

Comment:
The `+ 500` prevents running multiple events within the same 500s time period as an event is scheduled to run, by removing this the code is relying on the behaviour described in the codex that is subject to change.

Reply:
I don't think this is accurate. All that `+ 500` does is change the timestamp from "now" to 500 seconds later than "now". It won't have any effect on running multiple events beyond what `wp_schedule_single_event()` does internally to avoid duplicates within 10 minutes. I don't see a need to delay the event by 500 seconds, which is why I removed that value (which I initially added to give me a buffer to validated that all the events were actually scheduled).

Reply:
Ok, happy with you having looked into it.